lib/rolodex/config.ex

```diff
@@ -85,8 +84,44 @@ defmodule Rolodex.Config do
 
   @spec new(list()) :: Rolodex.Config.t()
   def new(kwl \\ []) do
-    struct(__MODULE__, kwl)
+    opts =
+      kwl
+      |> Map.new()
+      |> set_writer_config()
+      |> set_pipelines_config()
+
+    struct(__MODULE__, opts)
+  end
+
+  defp set_writer_config(opts), do: Map.put(opts, :writer, get_writer_config(opts))
+
+  defp get_writer_config(%{writer: writer}), do: WriterConfig.new(writer)
+  defp get_writer_config(_), do: WriterConfig.new()
+
+  defp set_pipelines_config(opts), do: Map.put(opts, :pipelines, get_pipelines_config(opts))
+
+  defp get_pipelines_config(%{pipelines: pipelines}) do
+    Map.new(pipelines, fn {k, v} -> {k, PipelineConfig.new(v)} end)
   end
+
+  defp get_pipelines_config(_), do: %{}
+end
+
+defmodule Rolodex.WriterConfig do
+  @moduledoc """
+  Defines writer config params.
+  """
+
+  defstruct file_path: "",
```

Comment:
can we provide a sensible default? `swagger.json` or something

Reply:
*`openapi.json`